### PR TITLE
updated with automations and setup.py

### PR DESCRIPTION
**Pull request by builderjer**
## Summary by CodeRabbit

- **New Features**
	- Introduced multiple GitHub Actions workflows to automate translation, build tests, license compliance checks, and publishing processes.
	- Added scripts for managing versioning and translations, enhancing the localization process.
  
- **Bug Fixes**
	- Improved handling of versioning and build processes to ensure smoother deployments.

- **Documentation**
	- Updated `.gitignore` to exclude build artifacts and metadata directories, improving repository cleanliness. 

- **Chores**
	- Added new scripts for version management and translation tasks, streamlining development workflows.

### CODE REVIEW

**Comment by builderjer** [comment]
Fixed most of the build tests, not sure I did them right, but it cant check the pypi yet.  Don't think I can fix that one?

File: .github/workflows/auto_translate.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  workflow_dispatch:
+  push:
+    branches:
+      - dev
+
+jobs:
+  translate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (mikejgray) [inline .github/workflows/auto_translate.yml:12]: Consider v4

File: .github/workflows/auto_translate.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          ref: dev
+          fetch-depth: 0 # otherwise, there would be errors pushing refs to the destination repository.
+      - name: Setup Python
+        uses: actions/setup-python@v1
```
Comment (mikejgray) [inline .github/workflows/auto_translate.yml:17]: Consider v4

File: .github/workflows/auto_translate.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          python scripts/translate.py
+      - name: Commit to dev
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Auto Translate
+          branch: dev
+  prepare_skillstore:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (mikejgray) [inline .github/workflows/auto_translate.yml:34]: v4

File: .github/workflows/auto_translate.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          ref: dev
+          fetch-depth: 0 # otherwise, there would be errors pushing refs to the destination repository.
+      - name: Setup Python
+        uses: actions/setup-python@v1
```
Comment (mikejgray) [inline .github/workflows/auto_translate.yml:39]: v4

File: .github/workflows/build_tests.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        run: |
+          pip install .
+      - uses: pypa/gh-action-pip-audit@v1.0.7
+        with:
+          # Ignore setuptools vulnerability we can't do much about
+          # Ignore numpy vulnerability affecting latest version for Py3.7
+          ignore-vulns: |
+            GHSA-r9hx-vwmv-q579
+            GHSA-fpfv-jqm9-f5jm
+            PYSEC-2022-43012
```
Comment (mikejgray) [inline .github/workflows/build_tests.yml:39]: Python 3.7 is EOL, is this still valid to ignore?

File: .github/workflows/dev2master.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+name: Push dev -> master
+on:
+  workflow_dispatch:
+
+jobs:
+  build_and_publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (mikejgray) [inline .github/workflows/dev2master.yml:11]: v4

File: __init__.py
Comment (mikejgray) [inline __init__.py:1]: Did you mean to add this back in, or is it just out of date with the target branch?

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+
+with open(path.join(BASE_PATH, "README.md"), "r") as f:
+    long_description = f.read()
+
+setup(
+    name=f"{SKILL_NAME}",
+    version=get_version(),
+    url=f"https://github.com/OpenVoiceOS/{SKILL_NAME}",
+    license="Apache2",
```
Comment (mikejgray) [inline setup.py:98]: Apache-2.0

**Review by mikejgray — CHANGES_REQUESTED** [review]
A few notes on actions versions, and a fix in setup.py. Also, I believe there's a more modern implementation of the publish workloads - see https://github.com/OpenVoiceOS/skill-alerts/tree/dev/.github/workflows for examples

**Comment by mikejgray** [comment]
Also, same comment about the py-audit step - is this valid to test when core requires a lower version of the libraries in question?

File: __init__.py
Comment (builderjer) [reply __init__.py:1]: Out of date.  Will drop

File: .github/workflows/auto_translate.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+on:
+  workflow_dispatch:
+
+jobs:
+  autotranslate:
+    env:
+      API_KEY: ${{secrets.DL_API_KEY}}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (mikejgray) [inline .github/workflows/auto_translate.yml:11]:
```suggestion
      - uses: actions/checkout@v4
```

**Review by mikejgray — CHANGES_REQUESTED** [review]
This won't work as-is - we either need to include the scripts folder or, preferably, use one of the shared automations at OpenVoiceOS/.github to handle publishing without the scripts.

File: .github/workflows/auto_translate.yml (hunk 8 above)
Comment (builderjer) [reply .github/workflows/auto_translate.yml:11]: On my phone. Make whatever changes that are needed

**Comment by emphasize** [comment]
About the repo name: I'd suggest changing the repo name to `ovos-skill-ip`

Just had a convo with @goldyfruit about that. we should have all in line
- repo name
- package name
- skill_id
Less confusing, less coding, less remembering ...

leaving as with a setup.py declaration
```python
URL = "https://github.com/OpenVoiceOS/ovos-skill-x"
SKILL_CLAZZ = ""

AUTHOR, SKILL_NAME = URL.split(".com/")[-1].split("/")
ADDITIONAL_AUTHORS = []
AUTHORS = ADDITIONAL_AUTHORS + [AUTHOR]

SKILL_PKG = SKILL_NAME.lower().replace('-', '_')
PLUGIN_ENTRY_POINT = f'{SKILL_NAME.lower()}.{AUTHOR.lower()}={SKILL_PKG}:{SKILL_CLAZZ}'
```

and (regarding the package name)
```python
setup(
    ...,
    package_dir={SKILL_PKG: ""},
    packages=[SKILL_PKG, ... ],
    package_data={SKILL_PKG: find_resource_files()},
    ...,
)
```

File: .github/workflows/build_tests.yml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install skill
+        run: |
+          pip install .
+      - uses: pypa/gh-action-pip-audit@v1.0.7
```
Comment (emphasize) [inline .github/workflows/build_tests.yml:32]: `pypa/gh-action-pip-audit@v1.0.8`

 pip-audit itself is pulling requests<x -> vuln with 1.07

**Review by emphasize — CHANGES_REQUESTED** [review]

File: .github/workflows/auto_translate.yml (hunk 8 above)
Comment (emphasize) [reply .github/workflows/auto_translate.yml:11]: > use one of the shared automations at OpenVoiceOS/.github

There are none. we should invest time into setting them up

File: .github/workflows/build_tests.yml (hunk 9 above)
Comment (builderjer) [reply .github/workflows/build_tests.yml]: fixed

**Comment by builderjer** [comment]
I have no issues with changing the repo name and such, but that can be a separate PR for the setup.py and other needed files after the repo itself is renamed?

File: .github/workflows/auto_translate.yml (hunk 8 above)
Comment (builderjer) [reply .github/workflows/auto_translate.yml:11]: I am not versed well enough for the automatons yet, shared ones would be great though.  Much easier to setup

**Comment by emphasize** [comment]
As is, the pypi name would be skill-ip

**Comment by JarbasAl** [comment]
we should start moving to the new automations https://github.com/OpenVoiceOS/.github/tree/feat/shared_actions1

and adopting https://openvoiceos.github.io/ovos-technical-manual/versioning/ as discussed

**Comment by builderjer** [comment]
fixed [here](https://github.com/OpenVoiceOS/ovos-skill-ip/commit/1fca8922471b7376388779cacf8f7babe765de16)

**Pull request closed without merging**